holidays/holiday_base.py

```diff
@@ -664,7 +664,8 @@ def __setitem__(self, key: DateLike, value: str) -> None:
             # If there are multiple holidays on the same date
             # order their names alphabetically.
             holiday_names = set(self[key].split(HOLIDAY_NAME_DELIMITER))
-            holiday_names.add(value)
+            for holiday_name in value.split(HOLIDAY_NAME_DELIMITER):
+                holiday_names.add(holiday_name)
```

Comment:
```suggestion
            holiday_names.update(value.split(HOLIDAY_NAME_DELIMITER))
```

Reply:
Looks much cleaner! There was an additional clean up logic originally. I should've refactored code after removing it.

Thanks 👍